### ISSUE TITLE
Crash clicking on popup X11

### ISSUE THREAD

**Issue by danwalmsley** [AvaloniaUI/Avalonia#2938]
https://youtu.be/cpAmJuYZYKo
@kekekeks suggested all platforms should be using Input priority for events.

**Comment by danwalmsley** [comment]
```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Avalonia.VisualTree.VisualExtensions.GetVisualsAt(IVisual visual, Point p, Func`2 filter)
   at Avalonia.Input.InputExtensions.GetInputElementsAt(IInputElement element, Point p)
   at Avalonia.Input.MouseDevice.SetPointerOver(IPointerDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties properties, KeyModifiers inputModifiers)
   at Avalonia.Input.MouseDevice.MouseMove(IMouseDevice device, UInt64 timestamp, IInputRoot root, Point p, PointerPointProperties properties, KeyModifiers inputModifiers)
   at Avalonia.Input.MouseDevice.ProcessRawEvent(RawPointerEventArgs e)
   at Avalonia.Input.InputManager.ProcessInput(RawInputEventArgs e)
   at Avalonia.X11.X11Window.<ScheduleInput>b__96_0()
   at Avalonia.Threading.JobRunner.RunJobs(Nullable`1 priority)
   at Avalonia.X11.X11PlatformThreading.HandleX11(CancellationToken cancellationToken)
   at Avalonia.X11.X11PlatformThreading.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at AvalonStudio.Shell.Shell.StartShellApp[TAppBuilder](TAppBuilder builder, String appName, ShellAppMainDelegate main, String[] args, IFactory layoutFactory) in /run/media/pascal/Shared/source/repos/AvalonStudio/AvalonStudio.Shell/src/AvalonStudio.Shell/Shell.cs:line 17
   at AvalonStudio.Program.Main(String[] args) in /run/media/pascal/Shared/source/repos/AvalonStudio/AvalonStudio/AvalonStudio/Program.cs:line 33
Aborted (core dumped)
```

**Comment by grokys** [comment]
@danwalmsley I don't think the NRE in your stacktrace would be called by event priority. What is the actual problem here?

**Comment by HendrikMennen** [comment]
I had this exception multiple times on Linux. It happens sometimes when a new Window/Popup is opened and the mouse hovers over it quickly. Looks like  something is not initialized once the window is visible giving a short timeframe to cause this crash

**Comment by danwalmsley** [comment]
Seems to be fixed by #3190
